homeassistant/components/cover/opengarage.py

```diff
@@ -0,0 +1,222 @@
+            self._name = DEFAULT_NAME
+            self._state = STATE_OFFLINE
+            self._available = False
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo cover."""
+        return True
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        data = {}
+
+        if self.signal is not None:
+            data[ATTR_SIGNAL_STRENGTH] = self.signal
+
+        if self.dist is not None:
+            data[ATTR_DISTANCE_SENSOR] = self.dist
+
+        if self._state is not None:
+            data[ATTR_DOOR_STATE] = self._state
+
+        return data
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        if self._state == STATE_UNKNOWN:
+            return None
+        else:
+            return self._state in [STATE_CLOSED, STATE_OPENING]
+
+    def close_cover(self):
+        """Close the cover."""
+        if self._state not in [STATE_CLOSED, STATE_CLOSING]:
+            self._state_before_move = self._state
+            self._state = STATE_CLOSING
+            ret = self._push_button()
+            return ret.get('result') == 1
+
+    def open_cover(self):
+        """Open the cover."""
+        if self._state not in [STATE_OPEN, STATE_OPENING]:
+            self._state_before_move = self._state
+            self._state = STATE_OPENING
+            ret = self._push_button()
+            return ret.get('result') == 1
+
+    def stop_cover(self):
+        """Stop the door where it is."""
+        self._state_before_move = None
+        self._state = STATE_STOPPED
+        ret = self._push_button()
+        return ret.get('result') == 1
+
+    def update(self):
+        """Get updated status from API."""
+        try:
+            status = self._get_status()
+            state = STATES_MAP.get(status.get('door'), STATE_UNKNOWN)
+            if self._state_before_move is not None:
+                if self._state_before_move != state:
+                    self._state = state
+                    self._state_before_move = None
+            else:
+                self._state = state
+
+            _LOGGER.debug("%s status: %s", self._name, self._state)
+            self.signal = status.get('rssi')
+            self.dist = status.get('dist')
+            self._available = True
+        except (requests.exceptions.ConnectionError,
+                requests.exceptions.ReadTimeout) as ex:
+            _LOGGER.error('Unable to connect to OpenGarage device: %(reason)s',
+                          dict(reason=ex))
+            self._state = STATE_OFFLINE
+        except KeyError as ex:
+            _LOGGER.warning('OpenGarage device %(device)s seems to be offline',
```

Comment:
A KeyError doesn't mean it's offline, it means that you requested a key on a dictionary that didn't hold that key. However, this can never happen because you use `.get()` which will never raise this error. You can remove this block.